infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/LogReplicationConfig.java

```diff
@@ -30,16 +55,21 @@
      */
     public LogReplicationConfig(Set<String> streamsToReplicate) {
         this.streamsToReplicate = streamsToReplicate;
+        this.maxNumMsgPerBatch = DEFAULT_MAX_NUM_MSG_PER_BATCH;
```

Comment:
It is harder for later code review. It is easier logic to init all things together in one place instead of jumping code around for simple logic.

Reply:
it is actually a good practice, to avoid code duplication, and having all inits in one single place or when a change comes, you'll have to change in several places (prone to errors):

https://stackoverflow.com/questions/7577627/purpose-of-constructor-chaining